pkg/controllers/networkpolicy/controller.go

```diff
@@ -764,3 +782,83 @@ func (r *NetworkPolicyReconciler) resetWatchers() error {
 func (r *NetworkPolicyReconciler) GetVirtualMachinePolicyIndexer() cache.Indexer {
 	return r.virtualMachinePolicyIndexer
 }
+
+// removeIndexerObjectsByAccount removes entries based on account, from all the np
+// controller indexers
+func (r *NetworkPolicyReconciler) removeIndexerObjectsByAccount(namespacedName string) error {
+	r.Log.Info("Clearing indexers for the account", "account", namespacedName)
+	atSgs, err := r.appliedToSGIndexer.ByIndex(addrAppliedToIndexerByAccountId, namespacedName)
+	if err != nil {
+		r.Log.Error(err, "failed to get appliedToGroups from indexer", "account", namespacedName)
+	}
+	if len(atSgs) == 0 {
+		r.Log.V(1).Info("No appliedToGroups found for account", "account", namespacedName)
+	}
+	for i := range atSgs {
+		if atSgs[i] == nil {
+			continue
```

Comment:
Why would this be nil?

Reply:
While testing out namespace deletion, I observed a crash saying `interface is nil`.
I observed it while traversing agSgs.

Reply:
When we retrieve objects using by Index, the length of the objects returned can be non-zero, yet while accessing the obj(interface), the obj(interface) can be nil. 

I tried this by deleting an object in the indexer and got all the objects using Byindex immediately. 
one of the object returned was nil.

So while retrieving objects by Index, we should check if the interface can be converted to a required object and then proceed.